crates/transaction-pool/src/pool/events.rs

```diff
@@ -5,10 +5,36 @@ use std::sync::Arc;
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+/// Wrapper around a transaction hash and the event that happened to it.
+#[derive(Debug)]
+pub struct TransactionEvent(TxHash, TransactionEventType);
+
+impl TransactionEvent {
+    /// Create a new transaction event.
+    pub fn new(hash: TxHash, event: TransactionEventType) -> Self {
+        Self(hash, event)
+    }
+
+    /// The hash of the transaction this event is about.
+    pub fn hash(&self) -> TxHash {
+        self.0
+    }
+
+    /// The event that happened to the transaction.
+    pub fn event(&self) -> &TransactionEventType {
+        &self.1
+    }
+
+    /// Split the event into its components.
+    pub fn split(self) -> (TxHash, TransactionEventType) {
+        (self.0, self.1)
+    }
+}
+
 /// Various events that describe status changes of a transaction.
 #[derive(Debug, Clone, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub enum TransactionEvent {
+pub enum TransactionEventType {
```

Comment:
I'd like to keep the name `TransactionEvent` and try to find another name for wrapper,

perhaps `PoolTransactionEvent` not sure, but this here is really the event